.github/workflows/build-and-test.yml

```diff
@@ -8,92 +8,166 @@ on:
         required: true
         type: string
 
-env:
-  DOCKER_BUILDER_IMAGE: mangatasolutions/node-builder:0.2
-  DISABLE_TTY: 1
+permissions:
+  contents: write
+  id-token: write
 
 jobs:
   build-node-image:
+    name: Build Docker image
     runs-on: [self-hosted, compile]
+    container:
+      image: mangatasolutions/node-builder:multi
     steps:
       - uses: actions/checkout@v3
-      - uses: docker/setup-buildx-action@v2.2.1
-      - uses: docker/login-action@v2.1.0
+      - name: Authenticate to Google Cloud
+        uses: google-github-actions/auth@v1
         with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          workload_identity_provider: '${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}'
+          service_account: '${{ secrets.GCP_SERVICE_ACCOUNT }}'
 
-      - name: Build and push
-        uses: docker/build-push-action@v3.2.0
+      - name: Cache the Cargo dependencies
+        uses: mansagroup/gcs-cache-action@v1.0.3
+        with:
+          bucket: mangata-node-ci-cache
+          path: |
+            ${{ github.workspace }}/target
+            /usr/local/cargo/git
+            /usr/local/cargo/registry
+            ~/.cache/sccache
+          key: compile-cache-${{ runner.os }}-${{ hashFiles('Cargo.lock') }}
+      
+      - name: Compile mangata-node code
+        run: cargo build --release --no-default-features --features=mangata-rococo,mangata-kusama,runtime-benchmarks
+      
+      - name: Build and push Docker image
+        run: |
+          docker login -u ${{ secrets.DOCKERHUB_USERNAME }} -p ${{ secrets.DOCKERHUB_TOKEN }}
+          docker build -f devops/dockerfiles/node-new/Dockerfile \
+          -t mangatasolutions/mangata-node:rococo-${{ inputs.version }} \
+          -t mangatasolutions/mangata-node:${{ inputs.version }} .
+          docker push mangatasolutions/mangata-node:${{ inputs.version }}
+          docker push mangatasolutions/mangata-node:rococo-${{ inputs.version }}
+      
+      - uses: actions/upload-artifact@v3
         with:
-          push: true
-          file: devops/dockerfiles/node-new/Dockerfile
-          cache-from: type=registry,ref=mangatasolutions/mangata-node
-          cache-to: type=registry,ref=mangatasolutions/mangata-node,mode=max
-          tags: |
-            "mangatasolutions/mangata-node:${{ inputs.version }}"
-            "mangatasolutions/mangata-node:rococo-${{ inputs.version }}"
-            "mangatasolutions/mangata-node:kusama-${{ inputs.version }}"
+          name: binary-and-wasms
+          path: |
+            target/release/mangata-node
+            target/release/wbuild/mangata-rococo-runtime/mangata_rococo_runtime.compact.compressed.wasm
```

Comment:
that would be great to have wasm stored as artifact but without the whole path and with some version postfix, maybe same as docker image `${{ inputs.version }}`

so instead of 
```
unzip wasms.zip
Archive:  wasms.zip
   creating: mangata-kusama-runtime/
   creating: mangata-rococo-runtime/
  inflating: mangata-kusama-runtime/mangata_kusama_runtime.compact.compressed.wasm  
  inflating: mangata-rococo-runtime/mangata_rococo_runtime.compact.compressed.wasm  
```

sth like 
```
wasms-6a486c4f5e227367f766fe89c5325a33ea6d1b0b/mangata_rococo_runtime.compact.compressed.wasm
wasms-6a486c4f5e227367f766fe89c5325a33ea6d1b0b/mangata_kusama_runtime.compact.compressed.wasm

```